pagerduty/resource_pagerduty_extension.go

```diff
@@ -0,0 +1,139 @@
+package pagerduty
+
+import (
+	"log"
+
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func resourcePagerDutyExtension() *schema.Resource {
+	return &schema.Resource{
+		Create: resourcePagerDutyExtensionCreate,
+		Read:   resourcePagerDutyExtensionRead,
+		Update: resourcePagerDutyExtensionUpdate,
+		Delete: resourcePagerDutyExtensionDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourcePagerDutyExtensionImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"endpoint_url": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"extension_objects": {
+				Type:     schema.TypeList,
```

Comment:
Are `extension_objects` ordered? If the ordering for these objects is relevant, then `TypeList` is correct and we need to ensure we're reading `extension_objects` and storing them in state as the API has them. 

If ordering is not relevant, consider using `TypeSet` instead.